impl/src/main/java/org/jboss/weld/annotated/slim/unbacked/UnbackedAnnotatedType.java

```diff
@@ -59,17 +59,17 @@ private UnbackedAnnotatedType(AnnotatedType<X> source, AnnotatedTypeIdentifier i
         for (AnnotatedConstructor<X> constructor : source.getConstructors()) {
             constructors.add(UnbackedAnnotatedConstructor.of(constructor, this, cache));
         }
-        this.constructors = immutableGuavaSet(constructors);
+        this.constructors = immutableSet(constructors);
```

Comment:
I'd use `org.jboss.weld.util.collections.ImmutableSet.Builder` here and below...